File: Golang/Minimum Height Triangle.go
/*
Given the base and area of a triangle, this challenge
is to find the height of the triangle.
*/

func main() {
    // Scan in the base and area of the triangle
    var a, b float64
    fmt.Scanf("%f %f", &b, &a)

    // The height is (2a)/b
    h := (2*a)/b

    // Print the height, rounded up to the nearest integer
    fmt.Println(int(math.Ceil(h)))
}

File: Java/Minimum Height Triangle.java
/*
Given the base and area of a triangle, this challenge
is to find the height of the triangle.
*/

static int lowestTriangle(int base, int area)
{
    // The height of a triangle is (2a)/b, rounded up
    // Doubles used for correct division
    double height = Math.ceil((2*area) / (double)base);

    // Print the height
    return (int)height;
}
